pkg/generate/code/check.go

```diff
@@ -132,37 +138,76 @@ func checkRequiredFieldsMissingFromShape(
 			missing = append(missing, primaryARNCondition)
 			continue
 		}
-		cleanMemberNames := names.New(memberName)
-		cleanMemberName := cleanMemberNames.Camel
 
-		resVarPath := koVarName
-		// Check that the field has potentially been renamed
-		renamedName, wasRenamed := r.InputFieldRename(
-			op.Name, memberName,
-		)
-		_, found := r.SpecFields[renamedName]
-		if found && !wasRenamed {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + cleanMemberName
-		} else if found {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + renamedName
-		} else {
-			_, found = r.StatusFields[memberName]
-			if !found {
+		resVarPath, err := getSanitizedMemberPath(memberName, r, op, koVarName)
+		if err != nil {
+			// If it isn't in our spec/status fields, we have a problem!
+			msg := fmt.Sprintf(
+				"GENERATION FAILURE! there's a required field %s in "+
+					"Shape %s that isn't in either the CR's Spec or "+
+					"Status structs!",
+				memberName, shape.ShapeName,
+			)
+			panic(msg)
+		}
+		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+	}
+
+	// Loop over the not-required member fields in the shape and determine
+	// if an ID field can be populated by an ID in either the Status or the Spec of the resource
+	// and generate an if condition checking for all required fields having non-nil
+	// corresponding resource Spec/Status values (required for ReadMany)
+	for field := range notRequiredFields {
+		crKind := strings.ToLower(r.Kind)
+		if strings.Contains(strings.ToLower(field), crKind) {
+			pluralize := pluralize.NewClient()
+			// CR shouldn't have plural of its kind as a field
+			singular := pluralize.Singular(field)
+
+			resVarPath, err := getSanitizedMemberPath(singular, r, op, koVarName)
+			if err != nil {
 				// If it isn't in our spec/status fields, we have a problem!
 				msg := fmt.Sprintf(
 					"GENERATION FAILURE! there's a required field %s in "+
 						"Shape %s that isn't in either the CR's Spec or "+
 						"Status structs!",
-					memberName, shape.ShapeName,
+					singular, shape.ShapeName,
 				)
 				panic(msg)
 			}
-			resVarPath = resVarPath + r.Config().PrefixConfig.StatusField + "." + cleanMemberName
+			missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+			// Not expecting multiple ID fields
+			break
```

Comment:
We can't be sure that the one we caught was the ID field, though? Some ReadMany calls may have multiple fields with the resource name (yep.) These are treated more like filters than selectors. I don't have any concrete examples off the top of my head, but my hypothetical would be `InstanceType` and `InstanceName`. We would want to catch `InstanceName` only. Filtering by `InstanceType` would narrow down results but not be useful for our `ReadOne` method.

I believe this is why in the `SetResourceIdentifiers`, we went with a list of expected field keys. `<Resource>Name`, `<Resource>ID`, `ID`, `Name`. We could recycle that list, just pluralising it?

Reply:
I see your point; I'll go with the list in next revision.

Related: did you see my bit about SecurityGroups in the PR description (at the bottom)? It seems like both GroupName and GroupId would be valid-- any thoughts there? I know when Setting Resource Identifiers we panic if there are multiple possibilities

Reply:
Agree with @RedbackThomson here. Instead of looping through the "non-required fields" like you have here, it may make more sense to simply search for `ID`, `Name`, `<Resource>ID` and `<Resource>Name` fields in the Spec or Status that have a matching field (or matching plural-named field) in the ReadMany Input shape.